### ISSUE TITLE
Fatal error during Maintenance mode: Sabre\DAV\Exception\ServiceUnavailable 

### ISSUE THREAD

**Issue by mwries** [nextcloud/server#6124]
### Steps to reproduce
During update NC 12.0.1 > 12.0.2 system is in Maintenance mode. Then I see 3 of these fatal errors. One (test) user is still logged in the website during Maintenance mode and clicking around in Files.

### Expected behaviour
No error

### Actual behaviour
During Maintenance mode I see 3 of these fatal errors:

Sabre\DAV\Exception\ServiceUnavailable: System in maintenance mode.
1.	[internal function] OCA\DAV\Connector\Sabre\MaintenancePlugin->checkMaintenanceMode(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
2.	/var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php - line 105: call_user_func_array(Array, Array)
3.	/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 466: Sabre\Event\EventEmitter->emit('beforeMethod', Array)
4.	/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php - line 254: Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))
5.	/var/www/nextcloud/apps/dav/appinfo/v1/webdav.php - line 71: Sabre\DAV\Server->exec()
6.	/var/www/nextcloud/remote.php - line 162: require_once('/var/www/nextcl...')
{main}

### General server configuration

**Operating system:** Linux nextcloud.systemsnavigator.com 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64

**Web server:** Apache/2.4.18 (Ubuntu) (apache2handler)

**Database:** pgsql PostgreSQL 9.5.7 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609, 64-bit

**PHP version:** 7.0.22-0ubuntu0.16.04.1
<details>
        <summary>PHP-modules loaded</summary>

```
 - Core
 - date
 - libxml
 - openssl
 - pcre
 - zlib
 - filter
 - hash
 - Reflection
 - SPL
 - session
 - standard
 - apache2handler
 - redis
 - PDO
 - xml
 - calendar
 - ctype
 - curl
 - dom
 - mbstring
 - fileinfo
 - ftp
 - gd
 - gettext
 - iconv
 - imap
 - intl
 - json
 - ldap
 - exif
 - mcrypt
 - pdo_pgsql
 - pgsql
 - Phar
 - posix
 - readline
 - shmop
 - SimpleXML
 - smbclient
 - sockets
 - sysvmsg
 - sysvsem
 - sysvshm
 - tokenizer
 - wddx
 - xmlreader
 - xmlwriter
 - xsl
 - zip
 - libsmbclient
 - Zend OPcache
```
</details>

### Nextcloud configuration

**Nextcloud version:** 12.0.2 - 12.0.2.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh install of NC12.0.0

**Where did you install Nextcloud from:**  VM

**Are you using external storage, if yes which one:** files_external is disabled

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** YOUR ANSWER HERE (LDAP/ActiveDirectory/Webdav/...)

<details>
        <summary>Signing status</summary>

```
[]
```
</details>

<details>
        <summary>Enabled apps</summary>

```
 - activity: 2.5.2
 - comments: 1.2.0
 - dav: 1.3.0
 - federatedfilesharing: 1.2.0
 - files: 1.7.2
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - issuetemplate: 0.2.2
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - nextcloud_announcements: 1.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - password_policy: 1.2.2
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - updatenotification: 1.2.0
 - user_ldap: 1.2.1
 - workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
 - admin_audit
 - encryption
 - federation
 - files_external
 - files_pdfviewer
 - files_videoplayer
 - firstrunwizard
 - gallery
 - survey_client
 - systemtags
 - user_external
```
</details>

<details>
        <summary>Content of config/config.php</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "192.168.11.131",
        "nextcloud.systemsnavigator.com"
    ],
    "datadirectory": "\/var\/ncdatamnt",
    "overwrite.cli.url": "https:\/\/nextcloud.systemsnavigator.com\/",
    "dbtype": "pgsql",
    "version": "12.0.2.0",
    "dbname": "nextcloud_db",
    "dbhost": "localhost",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "oc6fxph0akpf",
    "mail_smtpmode": "smtp",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": true,
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0,
        "dbindex": 0,
        "password": "***REMOVED SENSITIVE VALUE***"
    },
    "htaccess.RewriteBase": "\/",
    "loglevel": 2,
    "log_type": "file",
    "logfile": "\/var\/ncdatamnt\/nextcloud.log",
    "logtimezone": "Europe\/Amsterdam",
    "maintenance": false,
    "mail_from_address": "support",
    "mail_smtpauthtype": "LOGIN",
    "mail_domain": "systemsnavigator.com",
    "mail_smtpauth": 1,
    "mail_smtphost": "smtp.office365.com",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpsecure": "tls",
    "ldapIgnoreNamingRules": false,
    "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
    "skeletondirectory": "",
    "updater.release.channel": "stable"
}
```
</details>

### Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063

**Operating system:** YOUR ANSWER HERE

**Comment by MorrisJobke** [comment]
This is basically the expected behaviour. I guess the only difference: we should not log this case.

@georgehrke @LukasReschke @rullzer @icewind1991 Easily possible to not log this specific exception?

**Comment by MorrisJobke** [comment]
> @georgehrke @LukasReschke @rullzer @icewind1991 Easily possible to not log this specific exception?

Looks like I found an easy way.

**Comment by MorrisJobke** [comment]
Fix is in #6132
